### PR TITLE
Gelato relayer scaffolding

### PR DESCRIPTION
**Pull request by tkporter**
* Successor of #776 - you can see a slightly ugly diff here https://github.com/abacus-network/abacus-monorepo/compare/webbhorn/gelato-PR-submitter-prep..trevor/gelato-PR-submitter-prep. I mostly just addressed the comments in that PR and did a little cleaning up
* This does *not* include the following, which will be added in later PRs:
   * Any related infrastructure to actually use Gelato in deployments
   * Actually implementing ForwardRequestOps / correct logic for actually dealing with the "wait queue"

### CODE REVIEW

**Comment by tkporter** [comment]
Btw plan to do some local testing tomorrow- have a feeling some of the config changes will need some changes to the infra

File: rust/agents/relayer/src/relayer.rs [hunk 1]
```diff
@@ -166,21 +182,29 @@ impl Relayer {
...
-                    signed_checkpoint_receiver.clone(),
-                    self.core.settings.inboxes[inbox_name].gelato_conf.clone(),
-                )
-            })
-            .collect();
+            watch::channel::<Option<MultisigSignedCheckpoint>>(None);
+
+        let inboxes = self.inboxes();
+
+        let mut tasks: Vec<Instrumented<JoinHandle<Result<()>>>> =
```
Comment (mattiecnvr) [inline rust/agents/relayer/src/relayer.rs:191]: Not just this line but some around here as well, is there any real benefit to specifying this type?

**Review by mattiecnvr — APPROVED** [review]

**Comment by mattiecnvr** [comment]
I am starting to see why Webb described himself as a professional plumber. Lol. This looks painful.

File: rust/agents/relayer/Cargo.toml [hunk 2]
```diff
@@ -6,21 +6,24 @@ edition = "2021"
 [dependencies]
 tokio = { version = "1", features = ["rt", "macros"] }
 config = "0.13"
+color-eyre = { version = "0.6", optional = true }
 serde = {version = "1.0", features = ["derive"]}
 serde_json = { version = "1.0", default-features = false }
 ethers = { git = "https://github.com/gakonst/ethers-rs", branch = "master" }
+ethers-contract = { git = "https://github.com/gakonst/ethers-rs", branch = "master", features=["legacy"] }
```
[the remark this answers is not in the document]
Comment (tkporter) [reply rust/agents/relayer/Cargo.toml:13]: Gonna try to merge this one first and review #934 today

File: rust/abacus-base/src/settings/chains.rs [hunk 3]
```diff
@@ -31,17 +31,12 @@ impl Default for ChainConf {
...
 pub struct GelatoConf {
-    /// Whether to use Gelato's Relay service for processing messages on inboxes.
-    pub enabled_for_message_submission: bool,
-    // Other gelato configuration options can go here, like
-    // max-in-flight submissions we're willing to try, perhaps
-    // the Ethereum-based chain_id (which may differ from
-    // abacus domain), number of retry attempts before giving up,
-    // etc.
+    /// Whether to use the Gelato Relay service for transactions submitted to the chain.
+    pub enabled: bool,
```
[the remark this answers is not in the document]
Comment (tkporter) [reply rust/abacus-base/src/settings/chains.rs:39]: yeah. I may change the way this is configured in a later PR, this feels a little awkward to me too

[the remark this answers is not in the document]
Comment (tkporter) [reply rust/abacus-base/src/settings/chains.rs:39]: e.g. we could consider an enum or something similar to the ChainConf

**Pull request merged**